xarray/core/dataset.py

```diff
@@ -1775,6 +1783,74 @@ def reindex(self, indexers=None, method=None, tolerance=None, copy=True,
         coord_names.update(indexers)
         return self._replace_vars_and_dims(variables, coord_names)
 
+    def interp(self, method='linear', assume_sorted=False, kwargs={},
+               **coords):
+        """ Multidimensional interpolation of Dataset.
+
+        Parameters
+        ----------
+        **coords : {dim: new_coordinate, ...}
+            Keyword arguments with names matching dimensions and values.
+            coords can be an integer, array-like or DataArray.
+            If DataArrays are passed as coords, their dimensions are used
+            for the broadcasting.
+        method: {'linear', 'nearest'} for multidimensional array,
+            {'linear', 'nearest', 'zero', 'slinear', 'quadratic', 'cubic'}
+            for 1-dimensional array.
+        assume_sorted: boolean, optional
+            If False, values of x can be in any order and they are sorted
+            first. If True, x has to be an array of monotonically increasing
+            values.
+        kwargs: dictionary, optional
+            Additional keyword passed to scipy's interpolator.
+
+        Returns
+        -------
+        interpolated: xr.Dataset
+            New dataset on the new coordinates.
+
+        Note
+        ----
+        scipy is required.
+
+        See Also
+        --------
+        scipy.interpolate.interp1d
+        scipy.interpolate.interpn
+        """
+        from . import missing
+
+        indexers_list = self._validate_indexers(coords)
+
+        obj = self if assume_sorted else self.sortby([k for k in coords])
+
+        def maybe_variable(obj, k):
+            try:
+                return obj._variables[k]
+            except KeyError:
+                return as_variable((k, range(obj.dims[k])))
```

Comment:
Do we have a cleaner solution to get variable corresponding to a dimension without coordinate?
I found `_get_virtual_variables`, but it looks overkill.